httpconnect/httpconnect.go

```diff
@@ -38,7 +38,24 @@ func New(opts *Options) filters.Filter {
 		}
 	}
 
-	return &httpConnectHandler{opts}
+	f := &httpConnectHandler{Options: opts}
+	f.ic = interceptor.New(&interceptor.Opts{
+		Dial:      f.dial,
+		GetBuffer: buffers.Get,
```

Comment:
http-proxy-lantern and flashlight both use buffer pools, which they can plug in here. Avoids unnecessary allocations.